opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -411,6 +411,104 @@ def __new__(cls, *args, **kwargs):
             raise TypeError("Span must be instantiated via a tracer.")
         return super().__new__(cls)
 
+    def __init__(
```

Comment:
Maybe construct the `_ReadWriteSpan` at the call site and only pass that one in to avoid duplicating the long parameter list.